### PR TITLE
Fix ios crash at sem_init fail with errno=78(function not implemented)

### PR DESCRIPTION
**Pull request by halx99**

### CODE REVIEW

File: common/threads.h [hunk 1]
```diff
@@ -16,8 +16,10 @@
 
 #if defined(__APPLE__)
 #include <AvailabilityMacros.h>
-#if (MAC_OS_X_VERSION_MIN_REQUIRED > 1050) && !defined(__ppc__)
+#include <TargetConditionals.h>
```
Comment (barracuda156) [inline common/threads.h:19]: I am not sure `TargetConditionals.h` exists in every MacOS SDK. This has to be verified, otherwise it will break the build.

**Comment by barracuda156** [comment]
@halx99 Allow me to test it, please (will do in an hour or so, once GCC build is complete and machine is free), but otherwise looks good.

File: common/threads.h (hunk 1 above)
Comment (halx99) [reply common/threads.h:19]: Yes, the `TargetConditionals.h` exists on all modern apple platforms

**Comment by barracuda156** [comment]
@halx99 Yes, your fix works fine on 10.6 PPC. Should be good to merge.

@kcat Turned out, inttypes fix in CMakeLists does not work: https://github.com/kcat/openal-soft/pull/851#issuecomment-1571810525
Sorry, I just assumed that time it will do what it is supposed to without rebuilding with it -_-

UPD. I know why it fails! You have used `cinttypes` header in the test, but source code uses `inttypes.h`. The first header does not need `__STDC_FORMAT_MACROS` with GCC, the second does. We should either change the test or the source.

**Comment by halx99** [comment]
@kcat This pr also ready to be merge.

**Comment by sgowen** [comment]
FWIW, I've just verified that the crasher on iOS is fixed if I apply the changes from this PR

**Comment by barracuda156** [comment]
@kcat Would you fix the header in the inttypes check or should I make a PR for that?

**Comment by kcat** [comment]
I think I'd prefer having the sources use `cinttypes` instead. I can do it.

**Comment by barracuda156** [comment]
@kcat Great, thank you!

**Pull request merged**